toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -16,41 +20,77 @@ var (
 	ErrBootGrubMkconfigGeneration = NewImageCustomizerError("Boot:GrubMkconfigGeneration", "failed to generate grub.cfg via grub2-mkconfig")
 )
 
+// BootConfigType represents the type of boot configuration used by the image.
+type BootConfigType string
+
+const (
+	// BootConfigTypeGrubMkconfig indicates the image uses grub-mkconfig to generate grub.cfg.
+	BootConfigTypeGrubMkconfig BootConfigType = "grub-mkconfig"
+	// BootConfigTypeGrubLegacy indicates the image uses a manually maintained grub.cfg.
+	BootConfigTypeGrubLegacy BootConfigType = "grub-legacy"
+	// BootConfigTypeUki indicates the image uses UKI without grub.cfg.
+	BootConfigTypeUki BootConfigType = "uki"
+)
+
 type BootCustomizer struct {
 	// The contents of the /boot/grub2/grub.cfg file.
 	grubCfgContent string
 
 	// The contents of the /etc/default/grub file.
 	defaultGrubFileContent string
 
-	// Whether or not the image is using grub-mkconfig.
-	isGrubMkconfig bool
+	// The type of boot configuration used by the image.
+	bootConfigType BootConfigType
 }
 
 func NewBootCustomizer(imageChroot safechroot.ChrootInterface) (*BootCustomizer, error) {
 	grubCfgContent, err := ReadGrub2ConfigFile(imageChroot)
-	if err != nil {
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
 		return nil, err
 	}
 
 	defaultGrubFileContent, err := readDefaultGrubFile(imageChroot)
-	if err != nil {
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
 		return nil, err
 	}
 
-	isGrubMkconfig := isGrubMkconfigConfig(grubCfgContent)
+	// Determine boot configuration type
+	bootConfigType := determineBootConfigType(grubCfgContent, imageChroot)
 
 	b := &BootCustomizer{
 		grubCfgContent:         grubCfgContent,
 		defaultGrubFileContent: defaultGrubFileContent,
-		isGrubMkconfig:         isGrubMkconfig,
+		bootConfigType:         bootConfigType,
 	}
 	return b, nil
 }
 
+func determineBootConfigType(grubCfgContent string, imageChroot safechroot.ChrootInterface) BootConfigType {
+	// If grub.cfg doesn't exist, check for UKI
+	if grubCfgContent == "" {
+		hasUkis, err := baseImageHasUkis(imageChroot.(*safechroot.Chroot))
+		if err == nil && hasUkis {
+			return BootConfigTypeUki
+		}
+		// No grub.cfg and no UKIs - default to grub-legacy (edge case)
```

Comment:
Previously we would error out for this edge case. So, I would report an error here.